File: regenerate_index.py
import os
import json
import re
from datetime import datetime

# Get all digest files
digest_dir = './data/digest'
files = []

for filename in os.listdir(digest_dir):
    if filename.endswith('.md'):
        filepath = os.path.join(digest_dir, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Extract frontmatter
        frontmatter_match = re.match(r'^---\s*\n([\s\S]*?)\n---\s*\n', content)
        if not frontmatter_match:
            print(f"⚠️ No frontmatter in {filename}")
            continue
        
        frontmatter = frontmatter_match.group(1)
        data = {'id': '', 'date': '', 'tags': [], 'sources': [], 'title': {'zh': '', 'en': ''}}
        
        for line in frontmatter.split('\n'):
            line = line.strip()
            if line.startswith('id:'):
                data['id'] = line.split(':')[1].strip().replace('"', '')
            elif line.startswith('date:'):
                data['date'] = line.split(':')[1].strip().replace('"', '')
            elif line.startswith('tags:'):
                tags = line.split(':')[1].strip()
                data['tags'] = json.loads(tags.replace("'", '"'))
            elif line.startswith('sources:'):
                sources = line.split(':')[1].strip()
                data['sources'] = json.loads(sources.replace("'", '"'))
            elif line.startswith('title:'):
                title = line.split(':')[1].strip()
                data['title'] = json.loads(title.replace("'", '"'))
        
        entry = {
            'id': data['id'],
            'date': data['date'],
            'file': filename,
            'title': data['title'],
            'tags': data['tags'],
            'sources_count': len(data['sources'])
        }
        files.append(entry)

# Sort by date descending
files.sort(key=lambda x: x['date'], reverse=True)

# Generate index
index = {
    'updated': datetime.now().isoformat().replace('Z', '+08:00'),
    'files': files
}

print(f"Generated index with {len(files)} files:")
for f in files:
    print(f"  {f['id']} - {f['file']}")

# Write index
with open('./data/digest/index.json', 'w', encoding='utf-8') as f:
    json.dump(index, f, ensure_ascii=False, indent=2)

File: regenerate_index.js
const fs = require('fs');
const path = require('path');

// Get all digest files
const digestDir = './data/digest';
const files = fs.readdirSync(digestDir)
  .filter(f => f.endsWith('.md'))
  .map(f => {
    const filePath = path.join(digestDir, f);
    const content = fs.readFileSync(filePath, 'utf8');
    
    // Extract frontmatter
    const frontmatterMatch = content.match(/^---\s*\n([\s\S]*?)\n---\s*\n/);
    if (!frontmatterMatch) {
      console.log(`⚠️ No frontmatter in ${f}`);
      return null;
    }
    
    const frontmatter = frontmatterMatch[1];
    const lines = frontmatter.split('\n');
    const data = { id: '', date: '', tags: [], sources: [], title: { zh: '', en: '' } };
    
    lines.forEach(line => {
      const trimmed = line.trim();
      if (trimmed.startsWith('id:')) {
        data.id = trimmed.split(':')[1].trim().replace(/"/g, '');
      } else if (trimmed.startsWith('date:')) {
        data.date = trimmed.split(':')[1].trim().replace(/"/g, '');
      } else if (trimmed.startsWith('tags:')) {
        const tags = trimmed.split(':')[1].trim();
        data.tags = JSON.parse(tags.replace(/'/g, '"'));
      } else if (trimmed.startsWith('sources:')) {
        const sources = trimmed.split(':')[1].trim();
        data.sources = JSON.parse(sources.replace(/'/g, '"'));
      } else if (trimmed.startsWith('title:')) {
        const title = trimmed.split(':')[1].trim();
        data.title = JSON.parse(title.replace(/'/g, '"'));
      }
    });
    
    return {
      id: data.id,
      date: data.date,
      file: f,
      title: data.title,
      tags: data.tags,
      sources: data.sources.length
    };
  })
  .filter(Boolean)
  .sort((a, b) => new Date(b.date) - new Date(a.date));

// Generate index.json
const index = {
  updated: new Date().toISOString().replace('Z', '+08:00'),
  files: files
};

console.log('Generated index with', files.length, 'files:');
files.forEach(f => console.log(`  ${f.id} - ${f.file}`));

fs.writeFileSync('./data/digest/index.json', JSON.stringify(index, null, 2) + '\n');